src/lib/machines/workProgress/genericSearchMachine.ts

```diff
@@ -0,0 +1,109 @@
+import { createMachine, send } from "xstate";
+import { assign } from "@xstate/immer";
+import { castDraft } from "immer";
+import { ClientError } from "graphql-request";
+import { Maybe } from "../../../types/sdk";
+import { GenericSearchService } from "../../services/genericSearchService";
+/**
+ * Context for Search Machine
+ */
+export interface SearchContext {
+  findRequest: any;
+  maxRecords?: number;
+  searchResult?: any;
```

Comment:
Instead of using `any` here, could it use a generic to keep type safety?

Reply:
Good point. I even thought about it while implementing. But the searchService and workProgressService are using completely different structures. So I was not sure how to keep a generic structure in place. It would be great to have a chat about this sometime later.